miio/tests/test_vacuums.py

```diff
@@ -0,0 +1,55 @@
+"""Test of vacuum devices."""
+from collections.abc import Iterable
+from typing import List, Sequence, Tuple, Type
+
+import pytest
+
+from miio.device import Device
+from miio.integrations.vacuum.roborock.vacuum import ROCKROBO_V1
+from miio.interfaces import VacuumInterface
+
+# list of all supported vacuum classes
+VACUUM_CLASSES: Tuple[Type[VacuumInterface], ...] = tuple(
+    cl for cl in VacuumInterface.__subclasses__()  # type: ignore
+)
+
+
+def _all_vacuum_models() -> Sequence[Tuple[Type[Device], str]]:
+    """:return: list of tuples with supported vacuum models with corresponding class"""
+    result: List[Tuple[Type[Device], str]] = []
+    for cls in VACUUM_CLASSES:
+        assert issubclass(cls, Device)
+        vacuum_models = cls.supported_models
+        assert isinstance(vacuum_models, Iterable)
+        for model in vacuum_models:
+            result.append((cls, model))
+    return result  # type: ignore
```

Comment:
Instead creating a a list for these, I think `yield (cls, model)` should work just fine? Just a random remark, this isn't a real issue though.

Reply:
I do not know whether this would work, but I do not see any issues with current code.

Reply:
I have a gut feeling that it should work, but it's not that terribly important and I don't want to hinder the progress for that.